### ISSUE TITLE
Default value when adding.

### ISSUE THREAD

**Issue by Alerinos** [KorzhCom/EasyData#99]
Cool idea is you can insert default values that are in the model. So if we have:
```cs
public string DateFormat { get; set; } = "DD-MM-YYYY";
```
We will get this result:
![image](https://user-images.githubusercontent.com/42449535/149644933-57605c6c-1222-4394-90c1-88c837336ad5.png)

_Originally posted by @Alerinos in https://github.com/KorzhCom/EasyData/issues/91#issuecomment-1013797263_

**Comment by Alerinos** [comment]
EasyData should distinguish the date format and adapt the display and management of the dates.

```cs
    [Column(TypeName = "Date")]
    public DateTime Age { get; set; } = DateTime.MinValue;    
```

Base:
![image](https://user-images.githubusercontent.com/42449535/149666490-0ac77654-655a-4894-90b5-d4f629ede47d.png)

EasyData:
![image](https://user-images.githubusercontent.com/42449535/149666507-cd4efc83-b319-4e44-8a81-c0e75ab10aeb.png)
![image](https://user-images.githubusercontent.com/42449535/149666524-0b5784a4-25bf-446e-bde3-899bbc28f1d6.png)

DataBase:
```
    [Age]               DATE             NOT NULL,
```
![image](https://user-images.githubusercontent.com/42449535/149666583-dcd4b0b0-0106-46e7-8686-1dc765c074a7.png)

**Comment by antifree** [comment]
Hello, 
regarding your problem about date. If you use .NET 6, you can replace DateTime to DateOnly type. 
We use CLR type as base, not database types

**Comment by Alerinos** [comment]
@antifree 
Command: 
```
Add-Migration Games_v1 -Project Lib -context GamesContext
```
```
The property 'Details.DateReleased' could not be mapped because it is of type 'DateOnly', which is not a supported primitive type or a valid entity type. Either explicitly map this property, or ignore it using the '[NotMapped]' attribute or by using 'EntityTypeBuilder.Ignore' in 'OnModelCreating'.
```
Unfortunately, ef and migration do not fully support this type. The use of DateTime remains :(

**Comment by Alerinos** [comment]
@antifree 
I also recommend adding a type for input in html.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input

int type number
string type text
etc...

**Comment by antifree** [comment]
Yes, regarding DateOnly and TimeOnly types, ef core SQL server provider has not been implemented support yet.
So you can use the other approach, or wait for version 1.4.2 where you will be able to configure type via annotations or fluent API (#101)

About input, it has been added for some types. Possibly number input type is missing

**Comment by antifree** [comment]
Hello, we have implement DefaultValue support (f9d45500facba40d02e1481cc0659613ea8aa714).

You can set up default value via:
1. DefaultValueAttribute annotation
```c#
   [DefaultValue("DD-MM-YYYY")]
   public string DateFormat { get; set; } = "DD-MM-YYYY";
```
2. EfCore model builder Fluent API 
```c#
protected override void OnModelCreating(ModelBuilder modelBuilder)
{
       modelBuilder.Entity<Country>()
            .Property(c => c.DateFormat)
               .HasDefaultValue("DD-MM-YYYY");
}
```
3. EasyData Fluent API
```c#
enpoints.MapEasyData(options => {
     options.UseDbContext<AppDbContext>(opts => {
           opts.CustomizeModel(model => {
                 model.Entity<Country>()
                    .Attribute(c => c.DateFormat)
                       .SetDefaultValue("DD-MM-YYYY");
           });
     });
});
```
Unfortunately, it is impossible to get initialization value without class instantiation, so I decided not to implement such an approach

**Comment by Alerinos** [comment]
Can't be done on the front side?

```html
<input type="text" id="DateFormat" name="DateFormat" value="@Culture.DateFormat">
```

This is a normal model, so you can view its value.
@antifree @korzh

**Comment by Alerinos** [comment]
While editing, the data from the database is displayed. The same can be done when adding a new record, except that instead of from the database, it is from the model.
![image](https://user-images.githubusercontent.com/42449535/149993086-0818131f-828d-47f1-b889-7b90acbec587.png)

**Comment by antifree** [comment]
> Can't be done on the front side?
> 
> ```
> <input type="text" id="DateFormat" name="DateFormat" value="@Culture.DateFormat">
> ```
> 
> This is a normal model, so you can view its value.

No, EasyData.JS abstracted from the Razor

**Comment by antifree** [comment]
> While editing, the data from the database is displayed. The same can be done when adding a new record, except that instead of from the database, it is from the model.

EasyData works the other way, it is not similar to Razor pages

**Comment by Alerinos** [comment]
```cs
		string testVariable = "value";
		string nameOfTestVariable = nameof(testVariable);
		Console.WriteLine($"{nameOfTestVariable} {testVariable}");
```
Create a json that will return the model name, variable and default value
example:
Get: /crud/getModel/Culture
Output:
```
"DateFormat": "DD-MM-YYYY",
"Test": "Value"
```

**Comment by antifree** [comment]
Why do additional requests if you have the possibility to set up default value in the model using one of 3 approaches?

**Comment by Alerinos** [comment]
If you are doing a new database then you can use the first example. If you have an old database, you must now add a parameter that is a duplicate of the value.

In my opinion, it is better to use a standard that is already built in. Additionally, if someone has several projects in one, he is addicted to install a library with a new attribute in each one.
![image](https://user-images.githubusercontent.com/42449535/150000267-c86711fa-95d2-4211-8fb2-a6cd02ab3825.png)

**Comment by antifree** [comment]
OK. I've figure out how to implement this possibility too fef91db052b039e7b057817f2c7fded5d16e50eb

**Comment by Alerinos** [comment]
They do not work with GUID and DateTime.
Can you take a look, is it possible to add a default value?

```cs
public Guid Author { get; set; } = Guid.Empty;  
public DateTime Created { get; set; } = DateTime.Now; 
```

**Comment by antifree** [comment]
Hello, yes there was a problem with DateTime. We will publish the fix with the next release.

About GUID (and other ref types), we have to ignore the assignment if the value equals the default value of the reference type. For GUID it is empty GUID, for Int32 is 0, for DateTime is DateTime.MinValue, etc. 

We can remove this condition, but in this case, we will always show the default values for non-nullable reference types, which is not quite good for me. @korzh, what do you think?
So for such a case, you can use EasyData Fluent API to set up such a default value

**Comment by antifree** [comment]
Hello, yes there was a problem with DateTime. We will publish the fix with the next release.

About GUID (and other ref types), we have to ignore the assignment if the value equals the default value of the reference type. For GUID it is empty GUID, for Int32 is 0, for DateTime is DateTime.MinValue, etc. 

We can remove this condition, but in this case, we will always show the default values for non-nullable reference types, which is not quite good for me. @korzh, what do you think?
So for such a case, you can use EasyData Fluent API to set up such a default value

**Comment by antifree** [comment]
Fixed in version 1.4.4
